app/views/unattended/provisioning_templates/finish/kickstart_default_finish.erb

```diff
@@ -17,19 +17,24 @@ oses:
   - package_upgrade: boolean (default=true)
   - salt_master: string (default=undef)
   %>
-<%= snippet_if_exists(template_name + " custom pre") -%>
-<% if @host.subnet.respond_to?(:dhcp_boot_mode?) -%>
-<%= snippet 'kickstart_networking_setup' %>
-service network restart
-<% end -%>
-
 <%
   rhel_compatible = @host.operatingsystem.family == 'Redhat' && @host.operatingsystem.name != 'Fedora'
+  os_major = @host.operatingsystem.major.to_i
   puppet_enabled = host_puppet_server.present? || host_param_true?('force-puppet')
   salt_enabled = host_param('salt_master') ? true : false
   chef_enabled = @host.respond_to?(:chef_proxy) && @host.chef_proxy
 %>
 
+<%= snippet_if_exists(template_name + " custom pre") -%>
+<% if @host.subnet.respond_to?(:dhcp_boot_mode?) -%>
+<%= snippet 'kickstart_networking_setup' %>
+<% if (rhel_compatible && os_major >= 8) -%>
```

Comment:
The minimal version of operating system should be `7` instead of `8`? The systemd was first available on EL7?

Reply:
well, the main change is regarding the move towards NetworkManager. The template worked perfectly on EL7 but doesn't work in EL8 because NetworkManager need to be restarted instead of the "network" (from old network-scripts).